### PR TITLE
[C++][Restbed/Pistache] Added fix for byte array

### LINKED ISSUE #747: [C++][Restbed-server] Compilation error from missing header "ByteArray.h"
##### Description
The Restbed server generated code includes a header file that does not exists (ByteArray.h):

From Frame.h:
```
#ifndef Frame_H_
#define Frame_H_

#include <string>
#include "ByteArray.h"
#include "Event.h"
#include <map>
#include <vector>
#include <memory>
```

##### openapi-generator version
3.2.1 (cloned the master branch)

##### OpenAPI declaration file content or url
https://raw.githubusercontent.com/deepviss/deepviss-server-example/master/src/main/resources/deep-viss.json

##### Command line used for generation
Inside cloned repository, after mvn clean package command:
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate /
-i deep-viss.json /
-g cpp-restbed-server /
-o ../openapi-restbed-server

##### Steps to reproduce
I used the following cmake file:
```
cmake_minimum_required(VERSION 3.10)
project(restbed_ex)

set(CMAKE_CXX_STANDARD 11)

file(GLOB_RECURSE src_files mode/*.cpp api/*.cpp)

add_executable(restbed_ex main.cpp ${src_files})
target_include_directories(restbed_ex PRIVATE /restbed/include/)
target_include_directories(restbed_ex PRIVATE model/)
target_include_directories(restbed_ex PRIVATE api/)
target_link_libraries(restbed_ex restbed)
```
Note: Assuming restbed/include is the folder with restbed include files (https://github.com/Corvusoft/restbed) and that you copied the json.hpp file from https://github.com/nlohmann/json to model folder

The content of main.cpp file is redundant as the compilation won't pass the line where "ByteArray.h" is included (that is model/Frame.cpp)

### PR DESCRIPTION
**Pull request by stkrwork**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
Fix #747
@ravinikam @fvarose @etherealjoy @MartinDelille

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestbedServerCodegen.java [hunk 1]
```diff
@@ -301,6 +304,8 @@ public String toDefaultValue(Schema p) {
                 return "0L";
             }
             return "0";
+        } else if (ModelUtils.isByteArraySchema(p)) {
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestbedServerCodegen.java]: Shouldn't it be the same like for string schema?
```C++
         if (ModelUtils.isStringSchema(p)) {
             return "\"\"";
```

Comment (stkrwork) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestbedServerCodegen.java]: yes, you are right. i copied it over from pistache. so pistache seems to also have an error then.

**Comment by etherealjoy** [comment]
#747 spec has a lot of complicated structs, like map of arrays. 
I guess they would need a subsequent fix as well later.

**Comment by stkrwork** [comment]
Fix confirmed by @vbs96

**Pull request merged**